jaxb/src/main/java/no/digipost/xml/bind/MarshallingCustomization.java

```diff
@@ -0,0 +1,109 @@
+
+import java.util.Set;
+
+/**
+ * Encapsulates customizing the {@link Marshaller} and {@link Unmarshaller} used for JAXB marshalling.
+ *
+ * @see MarshallerCustomizer
+ * @see UnmarshallerCustomizer
+ */
+public final class MarshallingCustomization {
+
+    public static MarshallingCustomization validateUsingSchemaResources(Set<String> schemaResources) {
+        return of(
+                MarshallerCustomizer.validateUsingSchemaResources(schemaResources),
+                UnmarshallerCustomizer.validateUsingSchemaResources(schemaResources));
+    }
+
+    public static MarshallingCustomization validateUsingSchema(Schema schema) {
+        return of(MarshallerCustomizer.validateUsingSchema(schema), UnmarshallerCustomizer.validateUsingSchema(schema));
+    }
+
+    public static MarshallingCustomization onUnmarshalling(UnmarshallerCustomizer customizer) {
+        return of(MarshallerCustomizer.NO_CUSTOMIZATION, customizer);
+    }
+
+    public static MarshallingCustomization onMarshalling(MarshallerCustomizer customizer) {
+        return of(customizer, UnmarshallerCustomizer.NO_CUSTOMIZATION);
+    }
+
+    public static MarshallingCustomization of(MarshallerCustomizer marshallerCustomizer, UnmarshallerCustomizer unmarshallerCustomizer) {
+        if (marshallerCustomizer instanceof NoCustomization && unmarshallerCustomizer instanceof NoCustomization) {
+            return NO_CUSTOMIZATION;
+        }
+        return new MarshallingCustomization(marshallerCustomizer, unmarshallerCustomizer);
+    }
+
+    public static final MarshallingCustomization NO_CUSTOMIZATION =
+            new MarshallingCustomization(MarshallerCustomizer.NO_CUSTOMIZATION, UnmarshallerCustomizer.NO_CUSTOMIZATION, "MarshallingCustomization.NO_CUSTOMIZATION");
+
+
+
+    private final MarshallerCustomizer marshallerCustomizer;
+    private final UnmarshallerCustomizer unmarshallerCustomizer;
+    private final String description;
+
+    private MarshallingCustomization(MarshallerCustomizer marshallerCustomizer, UnmarshallerCustomizer unmarshallerCustomizer) {
+        this(marshallerCustomizer, unmarshallerCustomizer, null);
+    }
+
+    private MarshallingCustomization(MarshallerCustomizer marshallerCustomizer, UnmarshallerCustomizer unmarshallerCustomizer, String description) {
+        this.marshallerCustomizer = marshallerCustomizer;
+        this.unmarshallerCustomizer = unmarshallerCustomizer;
+        this.description = description;
+    }
+
+    public MarshallingCustomization andThen(MarshallingCustomization nextCustomization) {
+        if (nextCustomization == NO_CUSTOMIZATION) {
+            return this;
+        } else if (this == NO_CUSTOMIZATION) {
+            return nextCustomization;
+        } else {
+            return of(
+                    this.marshallerCustomizer.andThen(nextCustomization.marshallerCustomizer),
+                    this.unmarshallerCustomizer.andThen(nextCustomization.unmarshallerCustomizer));
+        }
+    }
```

Comment:
Dette skjønte jeg ikke helt 😛 Hvis man sender inn `NO_CUSTOMIZATION` ignorerer vi det? Men hvis `NO_CUSTOMIZATION` allerede er satt (som er default) så endrer vi. Og hvis man sender inn flere etter det legger vi det til? Så man kan ha flere customizere?

Reply:
Det er litt ekstra greier for å sørge for å ikke utføre `NO_CUSTOMIZATION`-customizeren flere ganger enn nødvending?

Reply:
Ja. Litt fancy-pants, men siden customizing gjøres _per_ marshalling/unmarshalling tenker jeg det var greit å unngå et "tomt" metodekall om mulig.